drivers/clock_control/clock_control_mcux_ccm.c

```diff
@@ -6,7 +6,9 @@
 
 #define DT_DRV_COMPAT nxp_imx_ccm
 #include <errno.h>
+#ifndef CONFIG_ARM64
```

Comment:
uhm, I don't think this is the right way to do this. Why do we need to include `soc.h` for ARM64 if the file is basically empty? If needed just move the header file that are currently in `soc.h` here.

Reply:
Hi, @carlocaione, the driver is used both for ARM32 and ARM64 platforms, there is still soc.h header file in ARM32 platforms, but currently we don't need it for ARM64, so the code only include soc.h for non ARM64 platforms, so we can make sure it works both for ARM32 and ARM64, not sure my understand is corret,thanks.

Reply:
oooh, ok, I see, didn't know it was still used by ARM32. @gmarull any preference here?

Reply:
the thing is, what does arm32 header provide that arm64 doesn't? is there any arm64 specific code in the driver? if not why is soc.h needed at all?

Reply:
I did a quick/dirty test on one of our Arm32 targets and this header isn't needed so I would remove it completely

Reply:
Thanks for your comments, have removed soc.h completely.